### DISCUSSION TITLE
Scanning "auto" update denied 3rd party mods

### DISCUSSION THREAD

**Discussion by JustSimplyKyle** [gorilla-devs/ferium#146] [Ideas]
Currently ferium just spits out an error if the mod had denied 3rd party access. I wonder is it possible to use the scanning function to check whether if the mod is up to date(still add it to the profile, and when doing `ferium upgrade` if the mod had denied 3rd party access, use this type of checking to check) . I don't know how restrictive the denial is, I hope it just disables the download url. If this is possible, it can *somewhat* retain the usability for ferium in these kind of mods.

**Comment by theRookieCoder** [comment]
> I hope it just disables the download url

Yes it _only_ nulls the download url.

I'm planning to add support for opening browser tabs (using [webbrowser](https://docs.rs/webbrowser)) with the download page of unsupported mods, then listening on the downloads folder (with [notify](https://docs.rs/notify)) and moving the files in to the output directory. This is what GDLauncher and ATLauncher are currently doing I believe. This way you will be able to add restricted mods too, and download them semi-automatically.

**Reply by JustSimplyKyle** [reply]
Notify! That's a really cool crate that I've never heard of. It seems like the perfect solution to this problem.

**Reply by JustSimplyKyle** [reply]
![image](https://user-images.githubusercontent.com/68589851/174440197-2ca1a0e4-be37-4af0-ae82-bbf8b949e197.png)
It seem slike it's possible to do it pure optionly(this is selenium python, but there are rust "binding" available too)
https://docs.rs/thirtyfour/latest/thirtyfour/
